gprofiler/merge.py

```diff
@@ -332,6 +332,38 @@ def _enrich_and_finalize_stack(
     return f"{enrich_data.application_prefix}{enrich_data.container_prefix}{stack} {count}"
 
 
+def concatenate_from_external_file(
+    collapsed_file_path: str,
+    container_names_client: Optional[ContainerNamesClient],
+) -> Tuple[str, str, str, int]:
+    """
+    Concatenate all stacks from all stack mappings in process_profiles.
+    Add "profile metadata" and metrics as the first line of the resulting collapsed file.
+    """
+    total_samples = 0
+    lines = []
+    start_time = None
+    end_time = None
+    # TODO: container names and application metadata
+    with open(collapsed_file_path) as file:
+        for index, line in enumerate(file):
+            if index == 0:
+                data = json.loads(line[2:])
```

Comment:
You should enrich the metadata here with the result of `get_current_metadata`. Values in `data` take precedence, so we need to deep merge `data` onto the resulting dict from `get_current_metadata`.